pkg/middleware/recovery/recovery_test.go

```diff
@@ -0,0 +1,210 @@
+		require.Equal(t, http.StatusText(http.StatusInternalServerError), strings.Trim(resp.Body.String(), "\n"))
+	})
+}
+
+func TestUnaryPanicInterceptor(t *testing.T) {
+	listner := bufconn.Listen(1024 * 1024)
+
+	serverOpts := []grpc.ServerOption{
+		grpc.ChainUnaryInterceptor(
+			[]grpc.UnaryServerInterceptor{
+				UnaryPanicInterceptor(logger.MustNewLogger("text", "info", "unix")),
+			}...,
+		),
+	}
+
+	t.Run("With Unary Panic Interceptor", func(t *testing.T) {
+		srv := grpc.NewServer(serverOpts...)
+		t.Cleanup(func() {
+			srv.Stop()
+		})
+
+		openfgav1.RegisterOpenFGAServiceServer(srv, &unimplementedOpenFGAServiceServer{})
+
+		go func() {
+			err := srv.Serve(listner)
+			if err != nil {
+				t.Fail()
+			}
+		}()
+
+		dialer := func(context.Context, string) (net.Conn, error) {
+			return listner.Dial()
+		}
+
+		ctx, cancel := context.WithCancel(context.Background())
+		t.Cleanup(func() {
+			cancel()
+		})
+
+		opts := []grpc.DialOption{
+			grpc.WithContextDialer(dialer),
+			grpc.WithTransportCredentials(insecure.NewCredentials())}
+
+		conn, err := grpc.DialContext(ctx, "", opts...)
+		require.NoError(t, err)
+
+		t.Cleanup(func() {
+			conn.Close()
+		})
+
+		cli := openfgav1.NewOpenFGAServiceClient(conn)
+
+		_, err = cli.Read(ctx, &openfgav1.ReadRequest{})
+		require.ErrorContains(t, err, "Read not implemented")
+
+		_, err = cli.Check(ctx, &openfgav1.CheckRequest{})
+		require.ErrorContains(t, err, http.StatusText(http.StatusInternalServerError))
+
+		_, err = cli.ReadAuthorizationModels(ctx, &openfgav1.ReadAuthorizationModelsRequest{})
+		require.ErrorContains(t, err, "method ReadAuthorizationModels not implemented")
+	})
+}
+
+func TestStreamPanicInterceptor(t *testing.T) {
+	listner := bufconn.Listen(1024 * 1024)
+	t.Cleanup(func() {
+		listner.Close()
+	})
+
+	serverOpts := []grpc.ServerOption{
+		grpc.ChainStreamInterceptor(
+			[]grpc.StreamServerInterceptor{
+				StreamPanicInterceptor(logger.MustNewLogger("text", "info", "unix")),
+			}...,
+		),
+	}
+
+	srv := grpc.NewServer(serverOpts...)
+	t.Cleanup(func() {
+		srv.Stop()
+	})
+
+	openfgav1.RegisterOpenFGAServiceServer(srv, &unimplementedOpenFGAServiceServer{})
+
+	go func() {
+		_ = srv.Serve(listner)
+	}()
+
+	dialer := func(context.Context, string) (net.Conn, error) {
+		return listner.Dial()
+	}
+
+	ctx, cancel := context.WithCancel(context.Background())
+	t.Cleanup(func() {
+		cancel()
+	})
+
+	opts := []grpc.DialOption{
+		grpc.WithContextDialer(dialer),
+		grpc.WithTransportCredentials(insecure.NewCredentials())}
+
+	conn, err := grpc.DialContext(ctx, "", opts...)
+	require.NoError(t, err)
+
+	cli := openfgav1.NewOpenFGAServiceClient(conn)
+	stream, err := cli.StreamedListObjects(ctx, &openfgav1.StreamedListObjectsRequest{})
+	require.NoError(t, err)
+
+	_, err = stream.Recv()
+	require.ErrorContains(t, err, http.StatusText(http.StatusInternalServerError))
+}
+
+type unimplementedOpenFGAServiceServer struct {
```

Comment:
You actually don't need this struct. The embeddeded struct `openfgav1.UnimplementedOpenFGAServiceServer` already implements the method stubs just the way you've done.

So just drop this code and use `openfgav1.UnimplementedOpenFGAServiceServer` directly.